operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractExternalDependentResource.java

```diff
@@ -106,4 +110,30 @@ public void handleDeleteTargetResource(P primary, R resource, String key, Contex
   protected InformerEventSource getExternalStateEventSource() {
     return externalStateEventSource;
   }
+
+  @Override
+  protected Optional<R> selectTargetSecondaryResource(
+      Set<R> secondaryResources, P primary, Context<P> context) {
+    R desired = desired(primary, context);
+    List<R> targetResources;
+    if (desired instanceof ExternalDependentIDProvider<?> desiredWithId) {
+      targetResources =
+          secondaryResources.stream()
+              .filter(
+                  r ->
+                      ((ExternalDependentIDProvider<?>) r)
+                          .externalResourceId()
+                          .equals(desiredWithId.externalResourceId()))
+              .toList();
+    } else {
+      throw new IllegalStateException(
+          "Either implement ExternalDependentIDProvider or override"
+              + " selectTargetSecondaryResource.");
```

Comment:
```suggestion
              + " AbstractDependentResource#selectTargetSecondaryResource.");
```

Reply:
I would appreciate to point where to find what I need to override but I'm ok with leaving it as is.